### ISSUE TITLE
How to remove 'ping' at first

### ISSUE THREAD

**Issue by sunjianan9900** [dvmatyun/web_socket#3]
![image](https://user-images.githubusercontent.com/10936316/197318570-7fe3f2c7-0d8b-46fd-bf36-e16c4a9d26b3.png)

  textSocketHandler.sendMessage(data1);
  textSocketHandler.sendMessage(data2);
  textSocketHandler.sendMessage(data3);

How can I disable sending "pings" to the server after a connection?
I don't need to send a "ping" to the server, which will result in an error return and denial of service by the server.
I have my own message to send, why do I need a "ping"?

**Comment by Lev09** [comment]
I’m facing the same issue. I think it would be great to be able to disable completely the ping/pong or be able to customize it to have different messages. 
In my case I would like to use 0/1 instead.

**Comment by dvmatyun** [comment]
@sunjianan9900 @Lev09 Does this solve your problem? (See full example at https://pub.dev/packages/websocket_universal #4 Advanced ISocketMessage handling. Composite server response example. But custom ping/pong messages I show below). If this is still not enough, I encourage you to see how `SocketMessageProcessor` class implements `IMessageProcessor<ISocketMessage<Object?>, IMessageToServer>` interface and then you can implement your own custom message handler, but I think example below will be sufficient for you. Let me know if you need any more help, feel free to message me using email **worc19966d@gmail.com** . And let me know if I can close the issue.

```
/// Postman echo ws server (you can use your own server URI)
  /// For local server it could look like 'ws://127.0.0.1:42627/websocket'
  const websocketConnectionUri = 'wss://ws.postman-echo.com/raw';
  const connectionOptions = SocketConnectionOptions(
    pingIntervalMs: 3000, // send Ping message every 3000 ms
    timeoutConnectionMs: 4000, // connection fail timeout after 4000 ms
    /// see ping/pong messages in logEventStream stream
    skipPingMessages: false,
  );

  /// Complex example:
  /// Example using [ISocketMessage] and [IMessageToServer]
  /// (recommended for applications, server must deserialize
  /// [ISocketMessage] serialized string to [ISocketMessage] object)

  /// Here I use custom ping/pong messages (they are the same for echo server), and they can be Byte too, but I don't have a 
  /// complete example
  final IMessageProcessor<ISocketMessage<Object?>, IMessageToServer>
      customPingProcessor = SocketMessageProcessor.single(
    'custom_ping_and_pong',
    'custom_ping_and_pong',
  );

  final socketHandler =
      IWebSocketHandler<ISocketMessage<Object?>, IMessageToServer>.createClient(
    websocketConnectionUri,
    customPingProcessor,
    connectionOptions: connectionOptions,
  );

  socketHandler.logEventStream.listen((logEvent) {
    // ignore: avoid_print
    print('> ws log: ${logEvent.data}, ping=${logEvent.pingMs}');
  });
```

**Comment by dvmatyun** [comment]
@sunjianan9900 @Lev09 Anyway, I published new version [websocket_universal: ^0.3.5] where `SocketConnectionOptions` has `skipPingMessages`. Set it to `true` if you want to ignore ping/pong message exchange. However I must warn you, that if you abort connection from server side - then the client will still think that it is connected (that is the behaviour of source google dart WS package). I will close this issue in a week if you have no more questions. **Please, leave a like for my package :)**
https://pub.dev/packages/websocket_universal

**Comment by dvmatyun** [comment]
Closed as completed and no more question were asked
